src/wallet/db.h

```diff
@@ -87,6 +87,33 @@ class CDBEnv
 
 extern CDBEnv bitdb;
 
+/** An instance of this class represents one database.
+ * For BerkeleyDB this is just a (env, strWalletFile) tuple.
+ **/
+class CWalletDBWrapper
+{
+public:
+    CWalletDBWrapper(CDBEnv *env_in, const std::string &strWalletFile_in):
+        env(env_in), strWalletFile(strWalletFile_in)
+    {
+    }
+
+    /** Rewrite the entire database on disk, with the exception of key pszSkip if non-zero
+     */
+    bool Rewrite(const char* pszSkip=nullptr);
+
+    /** Back up the entire database to a file.
+     */
+    bool Backup(const std::string& strDest);
+
+    /** Get a name for this database, for debugging etc.
+     */
+    std::string GetName() { return strWalletFile; }
```

Comment:
`std::string GetName() const`

You have this function, and the public `strWalletFile`, which you access directly in other places. Maybe `strWalletFile` should be made private?